src/utils/arfs_builders/arfs_builders.ts

```diff
@@ -25,7 +25,7 @@ export abstract class ArFSMetadataEntityBuilder<T extends ArFSEntity> {
 	constructor(protected readonly entityId: EntityID, protected readonly arweave: Arweave) {}
 
 	abstract getGqlQueryParameters(): GQLTagInterface[];
-	protected abstract buildEntity(): Promise<T>;
+	protected abstract buildEntity(driveKey?: DriveKey): Promise<T>;
```

Comment:
This is leaking subtype implementation details. We can't do this. The drive keys should be passed to the constructors of private subtypes only and cached for later usage.